### PR TITLE
Butter bar alerts! 🧈

### PR DESCRIPTION
**Pull request by lesleyjanenorton**
Here we:
- Add butter bar lit component 
- Add ButterBarService which manages alerts that should be shown in the UI 

New strings in[ this PR](https://github.com/mozilla-l10n/mozilla-vpn-extension-l10n/pull/2)

https://github.com/user-attachments/assets/a5517eaf-11ea-4a90-87ef-eb9ba416871b

### CODE REVIEW

File: src/background/butterBarService.js [hunk 1]
```diff
@@ -0,0 +1,129 @@
...
+    return dismissedAlerts.some((alertId) => alertId == id);
+  }
+
+  /**
+   * @param {string} id
+   * @param {Array} butterBarList
+   */
+  alertInButterBarList(id, butterBarList) {
+    return butterBarList.some((alert) => alert.alertId == id);
+  }
```
Comment (strseb) [inline src/background/butterBarService.js:105]: Both can be static :) making an easy test func.

File: src/background/butterBarService.js [hunk 2]
```diff
@@ -0,0 +1,129 @@
...
+
+  removeAlert(id) {
+    const newAlertList = this.butterBarList.value.filter(
+      ({ alertId }) => alertId !== id
+    );
+    this.dismissedAlerts.push(id);
+    return this.butterBarList.set(newAlertList);
```
Comment (strseb) [inline src/background/butterBarService.js:112]: `property.set` returns void. Should this return a value?

File: src/background/conflictObserver.js [hunk 3]
```diff
@@ -41,6 +41,10 @@ export class ConflictObserver {
    * @returns {boolean}
    */
   static isConflicting(addon) {
-    return addon.enabled && addon.permissions.includes("proxy");
+    return (
+      addon.enabled &&
+      addon.permissions.includes("proxy") &&
+      addon.id !== "vpn@mozilla.com"
```
Comment (strseb) [inline src/background/conflictObserver.js:47]: q__q oh wow how could i miss that.

**Review by strseb — APPROVED** [review]

File: src/background/butterBarService.js [hunk 4]
```diff
@@ -0,0 +1,129 @@
...
+
+export class ButterBarAlert {
+  /**
+   * @param {string} alertId
+   * @param {string} alertMessage
+   * @param {string} linkText
+   * @param {string} linkUrl
+   */
+  constructor(alertId, alertMessage, linkText, linkUrl) {
+    (this.alertId = alertId),
```
[the remark this answers is not in the document]
Comment (lesleyjanenorton) [reply src/background/butterBarService.js:124]: Agree - this was an `npm run format` change

**Pull request merged**